sql/hive/src/main/scala/org/apache/spark/sql/hive/MetastoreRelation.scala

```diff
@@ -49,8 +49,7 @@ private[hive] case class MetastoreRelation(
   override def equals(other: Any): Boolean = other match {
     case relation: MetastoreRelation =>
       databaseName == relation.databaseName &&
-        tableName == relation.tableName &&
-        output == relation.output
+        tableName == relation.tableName
```

Comment:
if the underlying Hive tables change the schema, I think we will break many places, as we cache the table relation. In the overwrite case, although the table schema changed, they are still the same table.